domain/modelagent/state/state_test.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state_test
+
+import (
+	"context"
+
+	"github.com/canonical/sqlair"
+	jc "github.com/juju/testing/checkers"
+	"github.com/juju/version/v2"
+	gc "gopkg.in/check.v1"
+
+	"github.com/juju/juju/core/model"
+	"github.com/juju/juju/domain"
+	modelerrors "github.com/juju/juju/domain/model/errors"
+	modelstatetesting "github.com/juju/juju/domain/model/state/testing"
+	modelagentstate "github.com/juju/juju/domain/modelagent/state"
+	schematesting "github.com/juju/juju/domain/schema/testing"
+)
+
+type stateSuite struct {
+	schematesting.ControllerSuite
+}
+
+var _ = gc.Suite(&stateSuite{})
+
+func (s *stateSuite) SetUpTest(c *gc.C) {
+	s.ControllerSuite.SetUpTest(c)
+}
+
+func (s *stateSuite) TestGetModelAgentVersionSuccess(c *gc.C) {
+	expectedVersion, err := version.Parse("4.21.54")
+	c.Assert(err, jc.ErrorIsNil)
+
+	txnRunner := s.TxnRunnerFactory()
+	state := modelagentstate.NewState(txnRunner)
+	modelID := modelstatetesting.CreateTestModel(c, txnRunner, "test")
+	s.setAgentVersion(c, modelID, expectedVersion.String())
+
+	obtainedVersion, err := state.GetModelAgentVersion(context.Background(), modelID)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(obtainedVersion, jc.DeepEquals, expectedVersion)
+}
+
+func (s *stateSuite) TestGetModelAgentVersionModelNotFound(c *gc.C) {
+	txnRunner := s.TxnRunnerFactory()
+	state := modelagentstate.NewState(txnRunner)
+
+	_, err := state.GetModelAgentVersion(context.Background(), "deadbeef")
```

Comment:
Can we please use valid model id's ? If you look at `github.com/juju/juju/core/model/testing` there is a func for generating them in tests.